test/benchmarks/src/main/scala/scala/collection/mutable/HashMapBenchmark.scala

```diff
@@ -0,0 +1,56 @@
+package scala.collection.mutable
+
+import org.openjdk.jmh.annotations._
+import org.openjdk.jmh.infra._
+import org.openjdk.jmh.runner.IterationType
+import benchmark._
+import java.util.concurrent.TimeUnit
+
+import scala.collection.mutable
+
+@BenchmarkMode(Array(Mode.AverageTime))
+@Fork(2)
+@Threads(1)
+@Warmup(iterations = 10)
+@Measurement(iterations = 10)
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+@State(Scope.Benchmark)
+class HashMapBenchmark {
+  @Param(Array("10", "100", "1000"))
+  var size: Int = _
+
+  var keys: List[Any] = _
+
+  @Setup(Level.Trial) def initKeys(): Unit = {
+    val alphabet = ('A' to 'Z')
+    keys = (0 to size).map(i => (i % 3) match {
+      case 0 => i
+      case 1 => alphabet(i % alphabet.size).toString
+      case 2 => i.toDouble
+    }).toList
+  }
+
+  var map = new HashMap[Any, Any]
+
+  @Setup(Level.Invocation) def initializeMutable = keys.foreach(v => map.put(v, v))
+
+  @TearDown(Level.Invocation) def tearDown = map.clear()
+
+  @Benchmark def getOrElseUpdate_found(bh: Blackhole): Unit = {
+    var i = 0
+    while (i < size) {
+      val value = map.getOrElseUpdate(i, -1)
+      bh.consume(value)
+      i += 1
+    }
+  }
+
+  @Benchmark def getOrElseUpdate_missing(bh: Blackhole): Unit = {
```

Comment:
these two (found and not found in map) should probably be separated, as they have different speeds.
Otherwise the result would depend on their cardinality's ratio.